src/crypto/index.js

```diff
@@ -93,6 +94,11 @@ function Crypto(baseApis, eventEmitter, sessionStore, userId, deviceId,
 
     this._globalBlacklistUnverifiedDevices = false;
 
+    // list of IncomingRoomKeyRequests/IncomingRoomKeyRequestCancellations
+    // we received in the current sync.
+    this._receivedRoomKeyRequests = [];
+    this._receivedRoomKeyRequestCancellations = [];
```

Comment:
E.g if I send a key request then immediately cancel it, currently we will process the key request due to how `_processReceivedRoomKeyRequests` is currently implemented.